### ISSUE TITLE
Chromebrew has no system to remove unused dependencies

### ISSUE THREAD

**Issue by saltedcoffii** [chromebrew/chromebrew#4572]
Chromebrew doesn't know if a dependency is unused and there's no system to remove unused dependencies. For example, I installed `rsync`. It called a dependency `xxhash`, however when running `crew remove rsync` it doesn't uninstall `xxhash`. It would greatly increase the usability of chromebrew if this feature was added, either with automatic unused dependency removals or with something like `crew remove --unused` (`crew remove -u`). There should also be a way to indicate to chromebrew whether a package is wanted, similar to portage's `--deselect` option or apt's `apt-mark`, for example `crew mark rsync --auto` or `crew mark xxhash --manual`. Installing a package when it's already installed should also mark a package that isn't manually installed as manually installed. Maybe even showing what's manually and automatically installed would be useful, for example `crew list auto` and `crew list manual`
I'm happy to clarify what I mean if I don't make sense.

**Comment by uberhacker** [comment]
Not sure what you mean by manually installed.  Unless metadata is available, there is no way to track manually installed "packages".  We can certainly improve the dependency management.  We just need a better dependency resolver.  You are welcome to look at the code and see how you could improve it.  We need somebody smart like you to step up and do it.  See #4338.

**Comment by vincowl** [comment]
@uberhacker Do you know if someone is already working on this ?

I think @saltedcoffii wants to have something equivalent to debian's apt system. Manually installed packages are the one that the user decided to install. In his example, that would be rsync. Other packages are declared as dependencies and removable if nobody needs it. These are removed using `apt autoremove` for instance.

**Comment by saltedcoffii** [comment]
Right. A manually installed package was explicitly installed by the user with a `crew install` command. An automatic package is one that was pulled in by a manually installed package.

**Comment by uberhacker** [comment]
@vincowl @saltedcoffii: AFAIK, there is nobody working on this ATM.  If you want to take a stab at this, that would be awesome.

**Comment by vincowl** [comment]
I'd really like to, but I am currently severely lacking time... Once I'll be over, I'll take a look at it. It would be great to have something like main use cases : how it should be run by the user and what services it should provide for instance.
